src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,192 @@
+
+    protected boolean isDone;
+
+    protected String type;
+
+    protected String userKeyedDateString;
+
+    protected java.time.LocalDate datetime;
+
+    /**
+     * Instantiates a new task.
+     *
+     * @param description the description
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * This constructor is used to retain most useful properties needed to create
+     * task at its core.
+     *
+     * @param description the description
+     */
+    public Task(boolean isDone, String description, String type) {
+        this.isDone = isDone;
+        this.description = description;
+        this.type = type;
+    }
+
+    /**
+     * Instantiates a new task.
+     *
+     * @param description the description
+     * @param datetime    the datetime
+     */
+    public Task(String description, String datetime) {
+        this.userKeyedDateString = datetime;
+        setDatetime(datetime);
+        this.description = description;
+    }
+
+    /**
+     * Instantiates a new task.
+     *
+     * @param isDone      the is done
+     * @param description the description
+     */
+    private Task(boolean isDone, String description) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    /**
+     * Sets the datetime.
+     *
+     * @param dt the new datetime
+     */
+    protected void setDatetime(String dt) {
+        assert (!dt.isEmpty());
+        dt = dt.replaceAll("\\s", "");
+        try {
+            java.time.format.DateTimeFormatter formatter
+                    = java.time.format.DateTimeFormatter.ofPattern("yyyy-MM-dd");
+            datetime = java.time.LocalDate.parse(dt, formatter);
+        } catch (Exception ex) {
+            try {
+                java.time.format.DateTimeFormatter formatter
+                        = java.time.format.DateTimeFormatter.ofPattern("dd/MM/yyyy hhmm");
+                datetime = java.time.LocalDate.parse(dt, formatter);
+            } catch (Exception ex2) {
+                System.out.println();
+            }
+
+        }
+    }
+
+    /**
+     * Prints the date time.
+     *
+     * @return the string
+     */
+    public String printDateTime() {
+        DateTimeFormatter outputformatter
+                = java.time.format.DateTimeFormatter.ofPattern("MMM d yyyy");
+        try {
+            String output = (datetime).format(outputformatter);
+            return output;
+        } catch (Exception ex) {
+            return userKeyedDateString;
+        }
+
+    }
+
+    /**
+     * Gets the status icon.
+     *
+     * @return the status icon
+     */
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718"); //return tick or X symbols
+    }
+
+    /**
+     * Checks if is done.
+     *
+     * @return true, if is done
+     */
+    public boolean isDone() {
+        return isDone;
+    }
+
+    /**
+     * Gets the description.
```

Comment:
It might be clearer if the javadoc comment explains what description the method is getting.